### ISSUE TITLE
It is possible to performance indexOf??

### ISSUE THREAD

**Issue by pedritin** [vadimg/js_bintrees#15]

**Comment by vadimg** [comment]
Yes, this is possible to do for a binary tree. Unfortunately, I don't have the time to implement every method someone requests, so if you want it, you'll have to add the method yourself and submit a pull request.

The indexOf (0-based index) of a node is `the size of the node's left subtree + (the parent's indexOf + 1, if the node is its parent's right child) - 1`

**Comment by vadimg** [comment]
To get an item at a specific index, you could do
```js
function getByIndex(i) {
  var item;
  var j = 0;
  tree.each(function(data) {
    if (j === i) {
      item = data;
      return false;
    }
    j++;
  });
  return item;
}
```

**Comment by vadimg** [comment]
However, if you are doing these types of operations often, I would suggest a built-in array (`[]`) would be much more performant and easier to use.
